src/Service.Tests/SqlTests/RestApiTests/RestBodyNonStrictModeTests/MsSqlRestBodyNonStrictModeTests.cs

```diff
@@ -276,7 +275,7 @@ public async Task PutOneWithRowversionFieldInRequestBody()
                     operationType: EntityActionOperation.Upsert,
                     requestBody: requestBody,
                     expectedStatusCode: HttpStatusCode.Created,
-                    expectedLocationHeader: expectedLocationHeader
+                    expectedLocationHeader: string.Empty
```

Comment:
I'm still a little confused from discussion in https://github.com/Azure/data-api-builder/pull/1731#discussion_r1332936049,
by setting `expectedLocationHeader: string.Empty` as empty string, we are no longer checking the expected location header. From the discussion, i see the reason is because the location header is incorrectly generated. As a result of that issue, is this test just ignoring the header or is the header returning as empty now until we fix the issue in a separate PR?

- if the intent is to no longer check the location header, because we are changing our contract of returning a location header, remove it as an argument completely. Also remove the assignment of the value `expectedLocationHeader = $"id/2";`

Reply:
Location header is made up of the following items : URL Request Scheme, URL Request Host, URL Request Path, Base Path (if configured) and primary key route computed by DAB.

Example values of each item (let's consider an example where base-route is not configured. There is no change to this discussion even if it is configured): 

URL Request Scheme: https
URL Request Host: localhost:5001
URL Request Path: This is different for POST vs PATCH/PUT requests. 
                               POST ---> api/commodities/. (Even in case of non-autogen PK, the PKs are specified in the request body)
                               PATCH/PUT ----> api/commoditites/categoryid/101/pieceid/101

Primary Key Route (which is computed by DAB):  /categoryid/101/pieceid/101. 

Since, the PK info is already present in the URL Request Path for PATCH and PUT requests, we shouldn't append the primary key route. Considering primary key route is not needed, it is not calculated as well. So, in tests, the right expected value would be an empty string.

But for POST requests, the PK info would not be present in the URL Request Path and therefore needs to be appended.

 
---> The logic for fixing this is present in this PR. When calling the `ConstructCreatedResponse()`, for PATCH/PUT requests, it is called with an empty string whereas for POST, with the primary key route. 
The tests are not ignoring the check for the primary key route. Rather, an empty string is the correct value.



Regarding your point about removing `expectedLocationHeader`, I thought I had removed all of them but seemed to have missed some. I will remove the remaining ones.